### ISSUE TITLE
Github Action relies on outdated flutter docker image

### ISSUE THREAD

**Issue by MrLightful** [alex-melnyk/flutter_advanced_avatar#11]
I see you're having some problems deploying the package.

Fyi, it seems [`sakebook/actions-flutter-pub-publisher`](url) is quite old and uses an unmaintained docker's cirrusci/flutter image repository, which provides Flutter 3.7.7 from 2 years ago. Cirrusci migrated all new images to a different place. 

I recommend finding a new Github Action that is up-to-date.

References:
https://github.com/sakebook/actions-flutter-pub-publisher/blob/master/Dockerfile#L1
https://hub.docker.com/r/cirrusci/flutter/tags

> DEPRECATED! Use here https://github.com/cirruslabs/docker-images-flutter/pkgs/container/flutter

**Comment by MrLightful** [comment]
@alex-melnyk try [firehose](https://github.com/dart-lang/ecosystem/tree/main/pkgs/firehose)

```
uses: dart-lang/ecosystem/.github/workflows/publish.yaml@main
with:
    use-flutter: true
````

https://github.com/alex-melnyk/flutter_advanced_avatar/commit/ec8817517bbb47efac5a30984d7551256e4f0bea#r149204162

**Comment by MrLightful** [comment]
@alex-melnyk I'm not sure if you turned off the action because you gave up on it, or for some other reason, but just to clarify in case you missed it: the latest version seemed to be working. It just [requested](https://github.com/alex-melnyk/flutter_advanced_avatar/actions/runs/11893608269/job/33138975208#step:6:195) your manual authorization to access pub.dev. I assume if you put permission tag, it would work automatically.

```
    permissions:
      id-token: write # Required for authentication using OIDC
```

But it's surely upto you how to publish. Was just curious to see if it was to work out.

**Comment by alex-melnyk** [comment]
> @alex-melnyk I'm not sure if you turned off the action because you gave up on it, or for some other reason, but just to clarify in case you missed it: the latest version seemed to be working. It just [requested](https://github.com/alex-melnyk/flutter_advanced_avatar/actions/runs/11893608269/job/33138975208#step:6:195) your manual authorization to access pub.dev. I assume if you put permission tag, it would work automatically.
> 
> ```
>     permissions:
>       id-token: write # Required for authentication using OIDC
> ```
> 
> But it's surely upto you how to publish. Was just curious to see if it was to work out.

Yes, it requested manual authorisation but redirect path is to localhost, so it doesn't work.
I assumed that it would work with credential.json instead of OIDC
